src/plugins/controls/public/control_types/options_list/options_list_editor.tsx

```diff
@@ -86,9 +86,10 @@ export const OptionsListEditor = ({
           onChangeDataViewId={(dataViewId) => {
             setLastUsedDataViewId?.(dataViewId);
             onChange({ dataViewId });
-            get(dataViewId).then((newDataView) =>
-              setState((s) => ({ ...s, dataView: newDataView }))
-            );
+            get(dataViewId).then((newDataView) => {
+              if (newDataView.id === dataView?.id) return;
+              setState((s) => ({ ...s, dataView: newDataView, fieldName: undefined }));
```

Comment:
**Nit:** I feel like because fetching the data view is async `fieldName` and `dataViewId` should be set  to undefined  **above** the `.then` callback. This is because if a `dataView`  was changed, the control editor would be in an erroneously valid state where the old field and data view is still selected until the new data view id is returned.

If their network was slow / it took long enough, they could potentially save and close the flyout and lose their change in the process.

Not the worst outcome here, but it could be cleaner to set everything to undefined first when it changes. There is a risk that this will cause a flicker on the field list though as it goes to an undefined state while the data view is loading - that's why this is just a nit / something to try!

Reply:
Great point! Addressed this in f95ce5c6a50a170bdd6032276ebfad3eb47c261a 👍